CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/BatchRun/BatchPostCoV2VTM.py

```diff
@@ -641,18 +707,115 @@ def load_rendering_manipulator(self, gd_manipulator, trial_idx=0, mcs=0):
         :param mcs: step at which to apply the manipulator
         :return: None
         """
-        if trial_idx not in self.__gd_manipulators.keys():
-            self.__gd_manipulators[trial_idx] = dict()
+        if trial_idx not in self._gd_manipulators.keys():
+            self._gd_manipulators[trial_idx] = dict()
+
+        self._gd_manipulators[trial_idx][mcs] = gd_manipulator
+
+    # todo - add API for defining rendering specs so users don't have to search through the details of GenericDrawer;
+    #  API should include convenience function for retrieving current specs
+    def load_screenshot_manipulator(self, sc_manipulator, trial_idx=0, mcs=0):
+        """
+        Loads a function for manipulating ScreenshotManagerCore rendering parameters at a simulation step of a trial
+        Manipulations are applied to all subsequent rendering processes
+        :param sc_manipulator: manipulator; signature should have argument of ScreenshotManagerCore object
+        :param trial_idx: trial at which to apply the manipulator
+        :param mcs: step at which to apply the manipulator
+        :return: None
+        """
+        if trial_idx not in self._sc_manipulators.keys():
+            self._sc_manipulators[trial_idx] = dict()
+
+        self._sc_manipulators[trial_idx][mcs] = sc_manipulator
+
+    def get_results_min_max(self, trial_idx):
+        """
+        Gets minimum and maximum over all simulation time for all available results
+        :return: {dict} range per available field
+        """
+        min_max_dict = dict()
+
+        self.load_trial_results(trial_idx)
+
+        if self.cml_results_reader is None:
+            print('No results loaded for trial {}.'.format(trial_idx))
+            return None
+
+        file_list = self.cml_results_reader.ldsFileList
+        for file_number, file_name in enumerate(file_list):
+            self.cml_results_reader.read_simulation_data_non_blocking(file_number)
+            sim_data_int_addr = extract_address_int_from_vtk_object(self.cml_results_reader.simulationData)
+            self.gd.field_extractor.setSimulationData(sim_data_int_addr)
+
+            for field_name, screenshot_data in self.scm.screenshotDataDict.items():
+                min_max = self._get_field_min_max(screenshot_data)
+                if min_max is not None:
+                    if field_name not in min_max_dict.keys():
+                        min_max_dict[field_name] = min_max
+                    else:
+                        if min_max[0] < min_max_dict[field_name][0]:
+                            min_max_dict[field_name][0] = min_max[0]
```

Comment:
Why not

`min_max_dict[field_name][0] = min(min_max_dict[field_name][0], min_max[0])`

?

Reply:
Either works